### ISSUE TITLE
"Fetch Failed" with certain images - not mod_security issue

### ISSUE THREAD

**Issue by diegovogel** [getgrav/grav-plugin-admin#1264]
The gallery page on one of my sites was empty this morning and the images also disappeared from the Page Media box in the Admin area. I got the "Fetch Failed" message when I opened the gallery page in Admin. I worked with my host's customer support and they confirmed there were no errors on their end, including anything related to mod_security.

The weird thing is that only certain images are causing this. If I FTP into the site and delete all the gallery images, then upload some random images that weren't there before through the Admin plugin, everything is fine. The new images display in Admin and on the website. But if I upload one of the images that was originally in the gallery it breaks again. I tried renaming the files, checked the file sizes, created a new page and uploaded the gallery images there, nothing works. For some reason these images have become unusable. 

The only error I'm getting consistently is this in the Dev Tools Console:

![image](https://user-images.githubusercontent.com/23545422/32614123-8bdb1400-c53a-11e7-8d5f-9c92db044d04.png)

I also got this Console error once right after uploading one of the cursed images but I haven't been able to reproduce it:
```
Uncaught TypeError: Cannot read property 'options' of undefined
    at n.value (admin.min.js?d817c0053e:5)
    at t.e.emit (vendor.min.js?d817c0053e:15)
    at t._finished (vendor.min.js?d817c0053e:16)
    at XMLHttpRequest.<anonymous> (vendor.min.js?d817c0053e:16)
```

**Comment by diegovogel** [comment]
UPDATE: I've reproduced this in a fresh install of Grav. I can save any image to Page Media except for the images I want to use for this gallery. When I try to save page with those images uploaded it gives me the "Fetch Failed" message and that same Console error. 

I tried an older version of the Admin plugin since the Console error seems to be related to some of the plugin JS files but that didn't help. Same problem. 

Really strange because that gallery was working fine until today. I don't know what else to try so any help is much appreciated!

**Comment by rhukster** [comment]
Can you send me the image files that cause the problem? rhuk at getgrav dot org

**Comment by diegovogel** [comment]
Here are a couple of them. 

![image-01](https://user-images.githubusercontent.com/23545422/32624250-abf57318-c556-11e7-8a82-6e51e2af5954.jpg)
![image-11](https://user-images.githubusercontent.com/23545422/32624251-ac07effc-c556-11e7-93c9-4ea0c4313eb1.jpg)

**Comment by rhukster** [comment]
The black and white one uploaded fine for me on my setup, and the second one gave an error:

https://cl.ly/3j0W0A0E210n/Image%202017-11-09%20at%2012.06.53%20PM.png

But this is due to my PHP memory limit being 2MB.  I think the issue might be with your server, perhaps a security rule that is limiting larger files.  Have you tried this on a local webserver setup?

**Comment by diegovogel** [comment]
Thanks for looking into this. Just to be clear, I can upload those images too. It's after uploading when I save the page and it reloads that I get the errors.

I don't think it's a file size limit because I can upload and save other images that are larger than these.

**Comment by rhukster** [comment]
Save works for me... I think that it probably is something related to a mod_security issue after all, and is probably directly related to the filename.  Try renaming the file and trying again?

**Comment by diegovogel** [comment]
Yeah I tried renaming the files. I even named other files the same name as the ones that don't work to see if they would cause problems, but they didn't. Somehow only those specific files are causing problems. I don't know what else the server would look at besides file name and file size to be able to distinguish these images from other ones.

**Comment by rhukster** [comment]
Have you double checked the mod_security logs?  what about the local setup or different host? That would definitely help isolate the issue to setup/configuration on the hosting side.

**Comment by diegovogel** [comment]
I'll try the local setup. Thanks for your help.

**Comment by diegovogel** [comment]
Tested this on a local server and had no issues. So I guess it's something with my host. When I talked to them earlier today they said nothing showed up in the error logs. Is there anything else it could be besides mod_security?

**Comment by diegovogel** [comment]
UPDATE (in case anyone with the same problem stumbles across this): I talked to my hosting provider again. They were not able to trace exactly what was causing this but they confirmed that the server somehow blacklisted these specific files. Renaming the files didn't help because the server was using a file ID number to identify the files, which doesn't change when you rename the file. Maybe the inode or something? I dunno. They said it may have something to do with mod_security rules but even if it did there wouldn't be anything they could do about it because I'm on shared hosting.

The only way I have found to get around this is to open the files in a photo editor and export them as a different resolution. Simply exporting doesn't do it, they need to be resized as well. 

`¯\_(ツ)_/¯`
